File: files/200-299/213/CH-213.py
import pandas as pd

path = "CH-213 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=16)
test = pd.read_excel(path, usecols="G:H", skiprows=1, nrows=3)

input['Group'] = pd.cut(input['Temperature'], bins=[-float('inf'), 10, 24, float('inf')], labels=['Cold', 'Mild', 'Hot'])
input['Group'] = input['Group'].astype(str)
result = input['Group'].value_counts().sort_values(ascending=True).reset_index().rename(columns={'count': 'No days'})

print(result.equals(test)) # True

File: files/200-299/213/CH-213.R
library(tidyverse)
library(readxl)

path = "files/CH-213 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:C18")
test  = read_excel(path, range = "G2:H5")

result = count(input, cut(Temperature, c(-Inf, 10, 24, Inf), c("Cold", "Mild", "Hot"))) %>%
  select(Group = 1, `No Days` = 2) %>%
  mutate(Group = as.character(Group))

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE